### PR TITLE
Cleanup duplications

### PR DESCRIPTION
**Pull request by RussTheAerialist**
Cleaning up the Endpoint base object and removing the duplicate ctors across all endpoints.

### CODE REVIEW

File: tableauserverclient/server/endpoint/auth_endpoint.py [hunk 1]
```diff
@@ -7,6 +7,9 @@
 
 
 class Auth(Endpoint):
+    def __init__(self):
+        super(Auth, self).__init__(None)
```
Comment (t8y8) [inline tableauserverclient/server/endpoint/auth_endpoint.py]: Keyword argument? I don't know what `None` means it kinda makes me think it's passing some fake value

Comment (t8y8) [inline tableauserverclient/server/endpoint/auth_endpoint.py]: Keyword argument? I don't know what None means it kinda makes me think it's passing some fake value

File: tableauserverclient/server/endpoint/endpoint.py [hunk 2]
```diff
@@ -8,64 +8,61 @@
...
-    def __init__(self):
-        self.parent_srv = None
+    def __init__(self, parent_srv):
+        self.parent_srv = parent_srv
 
     @staticmethod
-    def _check_status(server_response):
-        if server_response.status_code not in Success_codes:
-            raise ServerResponseError.from_response(server_response.content)
+    def _make_headers(token, content_type):
```
Comment (t8y8) [inline tableauserverclient/server/endpoint/endpoint.py]: `_make_tableau_headers` or `_make_server_headers` or something to indicate it's not a totally general purpose function, it really is Tableau REST API specific headers

File: tableauserverclient/server/endpoint/endpoint.py [hunk 3]
```diff
@@ -8,64 +8,61 @@
...
+        parameters['headers'] = Endpoint._make_headers(token, content_type)
+
+        if content is not None:
+            parameters['data'] = content
+
+        server_response = method(url, **parameters)
         self._check_status(server_response)
+
+        # This check is to determine if the response is a text response (xml or otherwise)
+        # so that we do not attempt to log bytes and other binary data.
```
Comment (t8y8) [inline tableauserverclient/server/endpoint/endpoint.py]: Thank you!!!

File: tableauserverclient/server/endpoint/endpoint.py [hunk 4]
```diff
@@ -8,64 +8,61 @@
...
-            raise ServerResponseError.from_response(server_response.content)
+    def _make_common_headers(token, content_type):
+        retval = {}
+        if token is not None:
+            retval['x-tableau-auth'] = token
+        if content_type is not None:
+            retval['content-type'] = content_type
 
-    def get_unauthenticated_request(self, url, request_object=None):
+    def _make_request(self, method, url, content=None, request_object=None, token=None, content_type=None):
```
Comment (RussTheAerialist) [inline tableauserverclient/server/endpoint/endpoint.py]: Okay

**Pull request merged**